openedx/core/djangoapps/catalog/docs/decisions/001-programs-cache.rst

```diff
@@ -0,0 +1,83 @@
+
+This cache is populated by calling LMS’s cache_programs management command.
+The cache is regularly refreshed via an external process.
+By design, the cache does not populate itself by calling out to the Discovery API upon a cache miss.
+
+Below is a (not-necessarily-exhaustive) list of features that depend on the programs cache:
+
+* The learner dashboard shows “Related Programs” under the cards of courses that are in programs.
+* The programs progress page uses the programs cache.
+* The creation of external IDs for users enrolled in courses within MicroBachelors programs requires knowledge of programs.
+* The program_enrollments system, which Masters uses for institution-managed enrollments in degree programs, uses the programs cache for validation of when creating associations between program enrollments and course enrollments.
+* The LMS management commands that populate program credentials need knowledge of programs.
+* Temporary Experiments often utilize the programs cache.
+* Courseware will reference programs information in order to show in-course recommendations.
+
+Issues
+======
+Correctness & Resiliency
+------------------------
+
+Because the cache is held in volatile memory, any crash or restart of the server holding the programs cache can surface these issues. While the cache_programs command can be run immediately afterwards, this is not a foolproof solution:
+
+* For a large course catalog, this command takes time to run, so there will be at least some time where LMS programs features are broken due to missing data.
+* Site operators may not know that this has to be done. Even if they do, it adds operational complexity.
+* The command may fail.
+
+Recently, the edX.org memcached cluster was restarted due to an increased LMS load. Following the restart, the cache_programs was not run immediately. As a result, there may have been an impact on LMS functionality due to the empty cache.
+
+Developer & operator experience
+=================================
+
+Requiring the regularly-scheduled execution of a management command to make the LMS fully functional adds operational complexity to courses.edx.org, Open edX instances, Devstack, and Sandboxes.
+
+Knowledge & access
+==================
+The requirements to populate the cache via the edx.org Jenkins job, the privileges required or the fact that the cache is not auto refreshed are not well known.
+
+Context boundaries
+==================
+It can be argued that LMS should only have access to data about course-runs, with logic around catalog-courses, curricula, programs, etc. all being handled by the IDAs that are meant to operate within those contexts (Discovery, Registrar, eCommerce, Credentials).
```

Comment:
I do not understand the meaning of this paragraph. Based on the argument here, what should be the correct cache implementation? Why a cache meant to serve LMS relates to all those other IDAs?